api/cluster/pke/leader_repository.go

```diff
@@ -104,10 +111,21 @@ func (r VaultLeaderRepository) SetLeader(organizationID, clusterID uint, leaderI
 	return err
 }
 
+func (r VaultLeaderRepository) DeleteLeader(organizationID, clusterID uint) error {
+	path := getMetadataPath(organizationID, clusterID)
+	_, err := r.logical.Delete(path)
```

Comment:
could we return 404 instead of 200 or 500 (I have no idea which one DELETE returns) when the path does not exist